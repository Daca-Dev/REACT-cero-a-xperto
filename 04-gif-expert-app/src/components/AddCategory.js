import React, { useState } from 'react';
import PropTypes from 'prop-types'


const AddCategory = ({ setCategories }) => {


    // se deben de usar useStates para poder cambiar y actualizar los valores de variables en el rende de react
    const [inputValue, setInputValue] = useState('');

    const handleInput = (event) => {

        setInputValue( event.target.value );

    }

    const handleSubmit = (event) => {
        /** manejar el evento de submit (presionar enter) en el formulario */
        // evita que el formulario recargue la página, esto ya no se debería aher así
        event.preventDefault();

        if ( inputValue.trim().length > 2 ) {
            // uso de la función pasada en los proptypes por medio de un callback
            setCategories( c => [ inputValue, ...c,] );
            setInputValue('');
        }
        
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInput }
                placeholder='Ingrese un texto de búsqueda...'
            />
        </form>
    )
}

// configuracion de los proptypes
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}


export default AddCategory
