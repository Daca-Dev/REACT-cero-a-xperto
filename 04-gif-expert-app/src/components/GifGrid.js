import React from 'react'
import PropTypes from 'prop-types'
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GifGrid = ({ category }) => {

    
    const { data: images, loading } = useFetchGifs( category );

    return (
        <div className="animate__animated animate__flipInX">
            <h3>{ category }</h3>
            <div className="cards-grid">
                { loading && <p>...Loading</p>}
                {
                    images.map( img => 
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    )
                }
            </div>
        </div>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}