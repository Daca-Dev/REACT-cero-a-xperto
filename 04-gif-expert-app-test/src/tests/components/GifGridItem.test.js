import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";


describe('Pruebas del componente <GifGridItem />', () => {

    const title = 'Hola mundo';
    const url = 'https://localhost/algo.png';
    const wrapper = shallow( <GifGridItem title={ title } url={ url }/>);
    
    test('Debe de mostrar el componente correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();

    })

    test('Debe de haber un parrafo con el contenido de title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })
    
    test('Debe de contener una imagen con la url', () => {
        const img = wrapper.find('img');
        // metodos para acceder a las propiedades del elemento
        // console.log( img.props());
        // console.log( img.prop('src'));
        expect( img.prop('src') ).toBe( url );
    })
    
    test('El contenedor principal debe de tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        // console.log(div.props());
        expect( div.prop('className').includes('animate__fadeIn') ).toBe( true );
        expect( div.prop('className').includes('animate__fadeIn') ).not.toBe( false );
    })
    
})
