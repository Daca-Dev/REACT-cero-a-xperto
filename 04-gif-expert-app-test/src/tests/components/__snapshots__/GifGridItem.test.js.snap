// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hola mundo"
    src="https://localhost/algo.png"
  />
  <p>
     
    Hola mundo
     
  </p>
</div>
`;
